src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleHeaderPanel.kt

```diff
@@ -91,6 +99,57 @@ class RuleHeaderPanel : JBPanel<RuleHeaderPanel>(FlowLayout(FlowLayout.LEFT)) {
     fun update(ruleKey: String, type: RuleType, severity: IssueSeverity) {
         clear()
         updateCommonFields(type, ruleKey)
+        updateRuleSeverity(severity)
+    }
+
+    fun update(project: Project, ruleKey: String, type: RuleType, severity: IssueSeverity, finding: Finding) {
+        clear()
+        updateCommonFields(type, ruleKey)
+        updateRuleSeverity(severity)
+        val serverFindingKey: String
+        val isIssueValid: Boolean
+        var taintVulnerability: LocalTaintVulnerability
+        var liveIssue: LiveIssue
+
+        when (finding) {
+            is LiveIssue -> {
+                serverFindingKey = finding.serverFindingKey ?: return
+                isIssueValid = finding.isValid
+                liveIssue = finding
+                listenIssueButton(isIssueValid, project, liveIssue.file, serverFindingKey, false,
+                    liveIssue, null)
+            }
+
+            is LocalTaintVulnerability -> {
+                serverFindingKey = finding.key() ?: return
```

Comment:
key for taint is not nullable
```suggestion
                serverFindingKey = finding.key()
```